worlds/Files.py

```diff
@@ -355,6 +356,8 @@ def write_token(self, token_type: APTokenTypes, offset: int, data: Union[bytes,
         """
         Stores a token to be used by patching.
         """
+        if not getattr(self, "tokens", None):
+            self.tokens = list()
```

Comment:
whu not in `__init__` ?

Reply:
I don't understand it well, but I was guessing it had something to do with `Mixin` being in the name, so multiple inheritance is expected, so `__init__` is likely to be missed.

I don't understand the intended usage of this class that well though.

Reply:
At least in my test using the KDL3 branch, `APTokenMixin.__init__` is missed if the subclass is also a ProcedurePatch. Doing it like this means we don't need to reach `__init__`.

Reply:
maybe this could be a safer way:

```python
    tokens: Sequence[
        Tuple[APTokenTypes, int, Union[
            bytes,  # WRITE
            Tuple[int, int],  # COPY, RLE
            int  # AND_8, OR_8, XOR_8
        ]]] = ()
```
```python
    def write_token(self, token_type: APTokenTypes, offset: int, data: Union[bytes, Tuple[int, int], int]):
        """
        Stores a token to be used by patching.
        """
        if not isinstance(self.tokens, list):
            self.tokens = []
        self.tokens.append((token_type, offset, data))
```

Reply:
maybe safer still:
```python
        if not isinstance(self.tokens, list):
            assert len(self.tokens) == 0, f"some funny business happened with {type(self)}.tokens"
            self.tokens = []
```
In case someone makes an assignment without `write_token`
(Should `write_token` be the only way that `tokens` gets modified?)

...also maybe renaming it to `_tokens`